### ISSUE TITLE
tourism=camp_site: power_supply asks wrong question

### ISSUE THREAD

**Issue by vsessink** [streetcomplete/StreetComplete#5776]
**How to Reproduce**
Find a tourism=camp_site or tourism=caravan_site. Wait for the question quest_camp_power_supply_title

The question says: "Is there any way for customers to charge their phone, laptop etc. here?" It then changes the power_supply tag.

This is wrong. The power_supply tag has nothing to do with _phones or laptops_.

The OSM tag "power_supply" just means that there is electricity "available" on the camp site: _power_supply=yes - means, there is a power supply available (e.g. on a camp_site)_ 

The tag seems to _imply_ that this means power supplies _at the pitches_, because a blue 3-pin socket is being described - which is the typical (European?) outdoor power supply for camp sites. (`power_supply cee_17_blue Blue 3-pin socket with 230V is available` is what the example says but  that is deprecated, `power_supply=yes` and `socket=cee_17_blue` would be the preferred way to encode a blue 3-pin socket).

**Expected Behavior**
Question should conform to the tag it is asking about, in this case: power_supply. As the camp_site OSM tag says that power_supply means "There is power supply available for customers", better wording would be: "is there electricity available for customers to use".

This _still_ is ambiguous, but more aligned to the OSM tag it changes.

"Is there electricity available on the pitches" or "are there power sockets available on the pitches" is probably the original meaning of the power_socket tag.

**External links**
https://wiki.openstreetmap.org/wiki/Proposal:Extend_camp_site camp_site extension proposals (with power_supply examples)
https://wiki.openstreetmap.org/wiki/Key:power_supply original power_supply tag
https://wiki.openstreetmap.org/wiki/Key:socket:* list of socket types

**Versions affected**
v58.w

**Comment by matkoniecz** [comment]
> power_supply tag has nothing to do with phones or laptops.

this is quite overstating things, as both are typically powered by electricity provided by power sockets

In which cases this question will result in wrong answers?

**Comment by matkoniecz** [comment]
maybe exact text was discussed at https://github.com/streetcomplete/StreetComplete/pull/4213

**Comment by vsessink** [comment]
Well, I'm actually at a camping place now. All pitches have a `cee_blue` socket, but my answer to "Is there any way for customers to charge their phone, laptop etc. here"  would definately have been "no", as there are no ways to charge anything here.

The only reason I answered "yes" was because I questioned the wording and took a search to see what the answer would do - which ended up being `power_supply=yes/no`.

Making the question specific (about "charging" and "phones") makes it confusing. A question should be as generic as possible. Making a question specific implies that the answer is specific, i.e. you don't ask "can you buy croissants here" if you just mean "is this a bakery" - and you don't ask "can you charge your phone here" if you mean "is there electricity for the customers".

The problem that `power_supply` seems to _imply_ that it's _on the pitches_ - but that that's not mentioned _in the definition_ is then left as an excercise to the dev team ;-)

**Comment by matkoniecz** [comment]
> took a search to see what the answer would do - which ended up being power_supply=yes/no.

BTW, exact tags are also shown in undo view (you can enable flight mode, do edit, undo edit and it will not even appear in database)

**Comment by matkoniecz** [comment]
@mnalis maybe you remember it being discussed one way or another?

**Comment by vsessink** [comment]
@matkoniecz here https://github.com/streetcomplete/StreetComplete/issues/4198#issuecomment-1183282143 and the reason for the specific question seems that @mnalis tries to ask for the minimum (i.e.  "is there _any sort of_ power available to customers, _at least_ to charge your phone"). But that, again, is not what is asked.

The issue that the power_supply tag itself is defined ambiguously in osm (acknowledged by @mnalis in https://github.com/streetcomplete/StreetComplete/issues/4198#issuecomment-1183575740) remains.

**Comment by westnordost** [comment]
So, I understand that this ambiguity could be solved by refining the wording? Does someone have a suggestion for a concise and unambiguous wording? The question should not be too long or complicated, otherwise people might skip on reading it. 

There is also the possibility to add an explanatory text behind an "info" button displayed to the right of the question (see [contributing a new quest](https://github.com/mnalis/StreetComplete/blob/mnalis-everything/CONTRIBUTING_A_NEW_QUEST.md#hint--hintimages-info-button)). However, it should only be used to explain doubts about (terms used in) the question itself, not mandate how the question should be answered, because people will only click that info button in the first place if they do have doubts and otherwise don't read it.

In any case, the quest is @mnalis work, so I'd like to leave dealing with enhancement suggestions to him. Let's see what he thinks about it.

**Comment by mnalis** [comment]
yes, it is not perfect, but seems to me the least confusing one. IIRC there was some back and forth both in the issue and the PR, and this was the one settled on.

The idea indeed was to provide whether there is power supply available for EVERYONE (and not just for caravans/RVs owners, which would likely often be misanswered as "no"  by most of them if the question was generic "is there electricity for the customers?" if the `cee_17_blue` socket was missing, but e.g. schuko was present at shared space)

> "is there any sort of power available to customers, at least to charge your phone"

might seem better - at least if one includes laptops specifically too, as sometimes usb-a might be provided (e.g. via solar-powered bench) but no regular domestic power connector like schuko, which again severly limits its usability for general use case (just like cee-17-blue does) -- but there are reasonable length limitations to observe too.


Nowdays quests IIUC allow to split the qestion into question+description, so perhaps it could be somewhat improved. Feel free to suggest how it might be improved in that way (I'll try to participate, but for next two weeks I might be quite limited in that)

Additional problem  for clarifications is that specific sockets available might vary by region,so that should be taken into consideration too.

**Comment by vsessink** [comment]
I'll send a patch with proposal when I'm home (currently still on a camping with eec_blue sockets, which started this all ;-)

**Comment by mnalis** [comment]
"Is a usable power supply available here?" and "Are there designated power sockets for guests to this campsite?" fail to clearly indicate "for who" - e.g. RV owner would likely answer exclusively on existence of `cee-17-blue` sockets (in most of  EU), which is not what we want. (see previous explanation)

_"Are there designated areas with power sockets for tents or caravans at this campsite?"_ is better in that regard (although IMHO still not  as explicit why we ask what we do), but it also introduces other confusions (e.g. in all smaller camps and most of the big ones IME there wouldn't be designated power sockets for tents, but they would exist in e.g. provided outdoor kitchen or covered "shared/mingle space" or elsewhere )

**Comment by vsessink** [comment]
Hoe about: "Is here electricity for guests"
Info box: "this can either be sockets at the pitches, or guest usable sockets in a service building"
Please note that the confusion in this discussion stems from the power_supply tag itself, which isn't clearly defined.

**Comment by rhhsm** [comment]
> Please note that the confusion in this discussion stems from the power_supply tag itself, which isn't clearly defined.

So maybe it should be defined clearly first (by starting a discussion on the community forum) and then implement it here?

**Comment by vsessink** [comment]
> > Please note that the confusion in this discussion stems from the power_supply tag itself, which isn't clearly defined.
> So maybe it should be defined clearly first (by starting a discussion on the community forum) and then implement it here?

Please see the **External links** above for a start. As it stands now, power_supply simply means "there is electricity usable for guests here". No definitions about at pitches or only central, paid or free, etcetera. The same goes for the streetcomplete question: it should only ask if power is available, nothing more, nothing less.

**Comment by mnalis** [comment]
**TL;DR:** I'd prefer status quo unless tag meaning is more clearly defined by community discussion, and/or some clear (to SC users!) but overlooked solution (that hasn't been suggested yet) is found. (also I might be very slow to respond for next week or two, sorry)

> What is your counter-proposal, or have I overlooked it?

@mcliquid  it was _"yes, it is not perfect, but seems to me the least confusing one."_ i.e. retain status quo (unless someone else can suggest idea which everybody agrees is **less** confusing than current one on all fronts)

@vsessink I guess "Is electricity generally available for all users here?" might be more to your liking? But I worry it might still be unclear for class of users that StreetComplete targets, see Quests guidelines below. Guiding them with clear example was the only way we could find workable at the time IIRC.

> The same goes for the streetcomplete question: it should only ask if power is available, nothing more, nothing less.

That would be next to useless (or worse), though. Technically the answer would be `yes` even if e.g. (to take the extreme) only solar-powered card payment terminal exists... Yes, you can add more words to clarify, but then there are other issues mentioned earlier too, and in trying to cover all those the question become too big and unwieldy, and unsuitable for SC on those grounds.

Having this one remain as-is, and having followup quest which asks for specific sockets available would clear that up, but that looks complex.

Note the SC [quest guidelines](https://github.com/streetcomplete/StreetComplete/blob/master/QUEST_GUIDELINES.md) though, especially _"👨‍💻 Users are no experts"_ and _"🐿️ Easy answer"_.

As for info text, see: 
> [...]  should only be used to explain doubts about (terms used in) the question itself, not mandate how the question should be answered

**Comment by rhhsm** [comment]
The problem with the present question is that it assumes that if there is electricity that guests can use, it is most likely that it is available through common standard sockets, for instance at a common service building, and these can be used to charge a phone or a laptop. In addition, there may be electricity available at pitches (all, or only some of them) through a special socket that is not suitable for charging phones. However, even in the less common case that there is electricity only at the pitches through a special socket, the answer to the question should be "Yes" (in my interpretation of the wiki) even though you can't charge phones at them. So I agree with @vsessink suggestion: "Is there electricity available for use by guests?" with info "This could be through sockets at the pitches, guest usable sockets in a service building, etc." (mentioning "sockets" makes it clear that it should be usable by at least some of the guests, who have the right plug to fit in the socket).

**Comment by vsessink** [comment]
How about:
`<string name="quest_camp_power_supply_title">"Are there power supply sockets available for guests?"</string>`
with a hint explaining:
`<string name="quest_camp_power_supply_hint">This can be sockets at the pitches, guest usable sockets in a service building, sockets for phone/laptop charging, etc.</string>`
(The hint would have to be added but as far as I understand the code that should be rather trivial).
